### ISSUE TITLE
Crash caused by NoSuchFieldError on AGP 8.3.0

### ISSUE THREAD

**Issue by JonatanPlesko** [infinum/android-sentinel#52]
## :writing_hand: Describe the bug
After updating my project to use AGP 8.3.0, the app doesn't start anymore, crashing with the following exception:

```
Caused by java.lang.NoSuchFieldError
No static field sentinel_crash_monitor of type I in class Lcom/infinum/sentinel/R$string; or its superclasses (declaration of 'com.infinum.sentinel.R$string' appears in /data/app/~~U6whZpJl9NNcgMuiTBeqbQ==/com.example.dev.debug-3QAT_NVWMtc0FLRVtLEGIA==/base.apk!classes17.dex)
    at com.infinum.sentinel.ui.tools.c.<init> (SourceFile:2)
    ...
```

## :bomb: Steps to reproduce
1. Update an app using Sentinel to use Android Gradle Plugin 8.3.0.
2. Attempt to start the app (make sure Sentinel.watch is called). 

## :wrench: Expected behavior
No crash 🙂

**Comment by KCeh** [comment]
OK I was able to reproduce crash:
```
 Caused by: androidx.startup.StartupException: java.lang.NoSuchFieldError: No static field sentinel_crash_monitor of type I in class Lcom/infinum/sentinel/R$string; or its superclasses (declaration of 'com.infinum.sentinel.R$string' appears in /data/app/~~dIyqmLflg1eB_ssGYVhreg==/com.infinum.template-J8U8hUulWm0MAuABqxfMEw==/base.apk!classes17.dex)
                                                                                                    	at androidx.startup.AppInitializer.doInitialize(AppInitializer.java:187)
                                                                                                    	at androidx.startup.AppInitializer.discoverAndInitialize(AppInitializer.java:238)
                                                                                                    	at androidx.startup.AppInitializer.discoverAndInitialize(AppInitializer.java:206)
                                                                                                    	at androidx.startup.InitializationProvider.onCreate(InitializationProvider.java:45)
                                                                                                    	at android.content.ContentProvider.attachInfo(ContentProvider.java:2482)
                                                                                                    	at android.content.ContentProvider.attachInfo(ContentProvider.java:2452)
                                                                                                    	at android.app.ActivityThread.installProvider(ActivityThread.java:7970)
                                                                                                    	at android.app.ActivityThread.installContentProviders(ActivityThread.java:7481) 
                                                                                                    	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7239) 
                                                                                                    	at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0) 
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2292) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:240) 
                                                                                                    	at android.os.Looper.loop(Looper.java:351) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8423) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013) 
                                                                                                    Caused by: java.lang.NoSuchFieldError: No static field sentinel_crash_monitor of type I in class Lcom/infinum/sentinel/R$string; or its superclasses (declaration of 'com.infinum.sentinel.R$string' appears in /data/app/~~dIyqmLflg1eB_ssGYVhreg==/com.infinum.template-J8U8hUulWm0MAuABqxfMEw==/base.apk!classes17.dex)
                                                                                                    	at com.infinum.sentinel.ui.tools.c.<init>(SourceFile:2)
                                                                                                    	at com.infinum.sentinel.di.a.<clinit>(Unknown Source:14)
                                                                                                    	at com.infinum.sentinel.SentinelInitializer.create(SourceFile:1)
                                                                                                    	at com.infinum.sentinel.SentinelInitializer.create(SourceFile:2)
                                                                                                    	at androidx.startup.AppInitializer.doInitialize(AppInitializer.java:180)
```

**Comment by KCeh** [comment]
This seems to be some sort of namespace collision. If you don't include other Sentinel tools (ex. Timber tool) then the reported error won't happen. 

But other issues can, like:

```
java.lang.IllegalStateException: WorkManager is already initialized.  Did you try to initialize it manually without disabling WorkManagerInitializer? See WorkManager#initialize(Context, Configuration) or the class level Javadoc for more information.
                                                                                                    	at androidx.work.impl.WorkManagerImpl.initialize(WorkManagerImpl.java:200)
                                                                                                    	at androidx.work.WorkManager.initialize(WorkManager.java:210)
                                                                                                    	at com.infinum.sentinel.ui.certificates.observer.b.<init>(Unknown Source:35)
                                                                                                    	at com.infinum.sentinel.di.component.b.C(Unknown Source:8)
```

I will look into R8 rules for Sentinel tools and AGP upgrade. That will hopefully resolve issues with AGP 8.3.0 compatibility.

**Comment by KCeh** [comment]
This should be resolved with [1.3.3-alpha01 ](https://github.com/infinum/android-sentinel/releases/tag/1.3.3-alpha01)
There is one drawback of the new version: you will need to enable [desugaring](https://developer.android.com/studio/write/java8-support#library-desugaring) in your project/module. We will try to improve this for full release.

**Comment by KCeh** [comment]
Full 1.3.3 release is [now out](https://github.com/infinum/android-sentinel/releases/tag/1.3.3)
No need to include desugaring in your project!
